request_test.go

```diff
@@ -1601,3 +1603,85 @@ func TestRequestErrorConflictMultipart(t *testing.T) {
 	req3.WithFileBytes("a", "a", []byte("a"))
 	req3.chain.assertFailed(t)
 }
+
+func TestRequestRedirect(t *testing.T) {
+	reporter := newMockReporter(t)
+
+	t.Run("dont follow redirects policy", func(t *testing.T) {
+		client := &http.Client{
+			Transport: mockTransportWithNoRedirect{t: t},
+		}
+
+		config := Config{
+			Client:   client,
+			Reporter: reporter,
+		}
+
+		req := NewRequest(config, "METHOD", "url").
+			WithHeader("Location", "/redirect").
+			WithRedirectPolicy(DontFollowRedirects)
+		req.chain.assertOK(t)
+
+		resp := req.Expect().
+			Status(http.StatusPermanentRedirect)
+		resp.chain.assertOK(t)
+
+		httpClient, _ := req.config.Client.(*http.Client)
+		assert.Equal(t, http.ErrUseLastResponse, httpClient.CheckRedirect(req.httpReq, nil))
+
+		assert.Nil(t, req.httpReq.GetBody)
+	})
```

Comment:
Nitpick: let's also check that tripCount has expected value (not zero), to be sure that our round tripper was really called.

Same for the second test.